core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -978,6 +979,26 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
+   * Add a directory to be downloaded with this Spark job on every node.
+   * The `path` passed must be a directory in HDFS (or other Hadoop-supported
+   * filesystems).  To access the directory in Spark jobs, use
+   * `SparkFiles.get(directoryName)` to find its download location.
```

Comment:
It would be good to say that the directory will be recursively copied into the root of Spark's scratch directory.